src/integration/java/io/pinecone/integration/controlPlane/pod/CollectionTest.java

```diff
@@ -129,18 +124,16 @@ public void testIndexFromDifferentMetricCollection() throws InterruptedException
         IndexMetric[] metrics = {IndexMetric.COSINE, IndexMetric.EUCLIDEAN, IndexMetric.DOTPRODUCT};
         IndexMetric targetMetric = IndexMetric.COSINE;
         for (IndexMetric metric : metrics) {
-            if (!metric.equals(indexMetric)) {
-                targetMetric = metric;
-            }
+            targetMetric = metric;
         }
 
         String newIndexName = RandomStringBuilder.build("from-coll", 8);
-        CreateIndexRequestSpecPod podSpec =
-                new CreateIndexRequestSpecPod().environment(environment).sourceCollection(collectionName);
-        CreateIndexRequestSpec spec = new CreateIndexRequestSpec().pod(podSpec);
-        createNewIndex(pineconeClient, newIndexName, dimension, targetMetric, spec, false);
+        createNewIndex(pineconeClient, newIndexName, dimension, targetMetric.toString(), false);
         indexesToCleanUp.add(newIndexName);
 
+        logger.info("Waiting until index is ready.");
+        waitUntilIndexIsReady(pineconeClient, newIndexName, 400);
+
```

Comment:
This test actually doesn't require the index be ready so we don't need to actually wait here, I think we can just remove this.

Reply:
Ah so I actually added that in bc on my local I thought it was timing out, but I know we were dealing w/a lot of errors w/tests yesterday, so I'll rerun to confirm.

Reply:
(Removed b/c changed test significantly)